tests/core/test_dataset.py

```diff
@@ -649,31 +649,50 @@ def test_set_active_scalars_name(grid):
 def test_rename_array_point(grid):
     point_keys = list(grid.point_data.keys())
     old_name = point_keys[0]
+    orig_vals = grid[old_name].copy()
     new_name = 'point changed'
     grid.set_active_scalars(old_name, preference='point')
     grid.rename_array(old_name, new_name, preference='point')
     assert new_name in grid.point_data
     assert old_name not in grid.point_data
     assert new_name == grid.active_scalars_name
+    assert np.array_equal(orig_vals, grid[new_name])
 
 
 def test_rename_array_cell(grid):
     cell_keys = list(grid.cell_data.keys())
     old_name = cell_keys[0]
+    orig_vals = grid[old_name].copy()
     new_name = 'cell changed'
     grid.rename_array(old_name, new_name)
     assert new_name in grid.cell_data
     assert old_name not in grid.cell_data
+    assert np.array_equal(orig_vals, grid[new_name])
 
 
 def test_rename_array_field(grid):
     grid.field_data['fieldfoo'] = np.array([8, 6, 7])
     field_keys = list(grid.field_data.keys())
     old_name = field_keys[0]
+    orig_vals = grid[old_name].copy()
     new_name = 'cell changed'
     grid.rename_array(old_name, new_name)
     assert new_name in grid.field_data
     assert old_name not in grid.field_data
+    assert np.array_equal(orig_vals, grid[new_name])
+
+
+def test_rename_array_dangling(grid):
+    # Regression test for issue #5244
+    def make_mesh():
+        m = pv.Sphere()
+        m.point_data['orig'] = np.ones(m.n_points)
+        return m
+
+    for _ in range(20):  # Depends on garbage collection
+        mesh = make_mesh()
+        mesh.rename_array('orig', 'renamed')
+        assert (mesh['renamed'] == 1).all()
```

Comment:
Can't we force garbage collection like [here](https://github.com/pyvista/pyvista/blob/19896236df625885e453f95677476ec588434354/tests/plotting/conftest.py#L45)?
E.g. something like
```suggestion
    mesh = make_mesh()
    mesh.rename_array('orig', 'renamed')
    gc.collect()
    assert (mesh['renamed'] == 1).all()
```

Reply:
You're right, it's not a great test. I think the issue is not whether gc runs, but whether the freed memory is overwritten so we can detect the change. I just changed the test to test directly that the array isn't deleted with a VTKObject.AddObserver('DeleteEvent') callback.